sklearn/preprocessing/tests/test_data.py

```diff
@@ -1987,6 +1992,49 @@ def test_one_hot_encoder_categorical_features():
     _check_one_hot(X, X2, cat, 5)
 
 
+@pytest.mark.parametrize(
+    "output_dtype",
+    [np.int32, np.float32, np.float64]
+)
+@pytest.mark.parametrize(
+    "input_dtype",
+    [np.int32, np.float32, np.float64]
+)
+@pytest.mark.parametrize(
+    "sparse",
+    [True, False]
+)
+def test_one_hot_encoder_preserve_type(input_dtype, output_dtype, sparse):
+    X = np.array([[0, 1, 0, 0], [1, 2, 0, 0]], dtype=input_dtype)
+    transformer = OneHotEncoder(categorical_features=[0, 1],
+                                dtype=output_dtype, sparse=sparse)
+    X_trans = transformer.fit_transform(X)
+    assert X_trans.dtype == output_dtype
+
+
+@pytest.mark.parametrize("output_dtype", [np.int32, np.float32, np.float64])
+@pytest.mark.parametrize("input_dtype", [np.int32, np.float32, np.float64])
+@pytest.mark.parametrize("sparse", [True, False])
+def test_one_hot_encoder_mixed_input_given_type(input_dtype, output_dtype,
```

Comment:
I'm tired, but it's not clear to me how this is distinct from above.

Reply:
Uhm, I did not see but but it has an unnecessary test.
The only test required was: https://github.com/scikit-learn/scikit-learn/pull/11042#discussion_r186405613